test/pylib/scylla_cluster.py

```diff
@@ -326,7 +326,7 @@ async def cql_is_up(self) -> bool:
                          protocol_version=4,
                          auth_provider=auth) as cluster:
                 with cluster.connect() as session:
-                    session.execute("SELECT * FROM system.local")
+                    session.execute("SELECT key FROM system.local where key = 'local'")
```

Comment:
If you don't add a comment here, I'll keep asking and forget what you answered: Why do you need to do an expensive SELECT (even if it's now less expensive) instead of a do-nothing CQL request? cql-pytest/conftest.py uses `BEGIN BATCH APPLY BATCH"`.

By the way, I'm not even sure why you need to do anything special after a successful connect(). A `connect()` should mean that the CQL connection is working. You may only need to test the CQL request after some time has passed (e.g., after a test), and this is what cql-pytest does.

Reply:
There is a comment above:
```
        # auth::standard_role_manager creates "cassandra" role in an
        # async loop auth::do_after_system_ready(), which retries
        # role creation with an exponential back-off. In other
        # words, even after CQL port is up, Scylla may still be
        # initializing. When the role is ready, queries begin to
        # work, so rely on this "side effect".
```

and `BEGIN BATCH APPLY BATCH` does not go through this authentication step, so we cannot rely on it.

Reply:
(I recall that we had this discussion already, @kostja checked whether `BEGIN BATCH APPLY BATCH` is enough, and it wasn't)

Reply:
```diff
diff --git a/test/pylib/scylla_cluster.py b/test/pylib/scylla_cluster.py
index 960879608..73004b654 100644
--- a/test/pylib/scylla_cluster.py
+++ b/test/pylib/scylla_cluster.py
@@ -334,6 +334,8 @@ class ScyllaServer:
                          auth_provider=auth) as cluster:
                 with cluster.connect() as session:
                     connected = True
+                    # See the comment above about `auth::standard_role_manager`. We execute
+                    # a 'real' query to ensure that the auth service has finished initialiing.
                     session.execute("SELECT key FROM system.local where key = 'local'")
                     self.control_cluster = Cluster(execution_profiles=
                                                         {EXEC_PROFILE_DEFAULT: profile},
```
@nyh would this be enough?

Reply:
Yes, thanks.